controllers/tasks.js

```diff
@@ -321,12 +321,28 @@ const updateTaskStatus = async (req, res, next) => {
     if (task.taskData.status === TASK_STATUS.VERIFIED || req.body.status === TASK_STATUS.MERGED)
       return res.boom.forbidden("Status cannot be updated. Please contact admin.");
 
+    if (userStatusFlag) {
+      if (task.taskData.status === TASK_STATUS.DONE && req.body.percentCompleted < 100) {
+        if (req.body.status === TASK_STATUS.DONE || !req.body.status) {
+          return res.boom.badRequest("Task percentCompleted can't updated as status is DONE");
+        }
+      }
+
+      if (
+        (req.body.status === TASK_STATUS.DONE || req.body.status === TASK_STATUS.VERIFIED) &&
```

Comment:
what if n/w failed and req came as null or empty

Reply:
already handled in middleware validation